### ISSUE TITLE
'llvm-c/Analysis.h' file not found

### ISSUE THREAD

**Issue by lucfis** [marcusrossel/kaleidoscope#2]
Upon running `swift build` I get the error `'llvm-c/Analysis.h' file not found`. I know that in the [blog post](http://marcusrossel.com/2020-06-02/kaleidoscope-3) you wrote you mentioned that Xcode was giving you this same error, but `swift build` was working for you. However, for me `swift build` is not working.

I'm using LLVM 16, and acknowledge that (as you mentioned in the blog post) some of the headers may be outdated, but `Analysis.h` is still present in LLVM 16.

I'm very very new to LLVM and Swift and therefore may have messed up somewhere in some obvious way. This may be a completely incorrect train of thought, but could it be that the "#include <llvm-c/Analysis.h>" is not *looking* in the correct place, and is therefore why the file can't be found. I say this because I definitely have the file stored at `/opt/homebrew/Cellar/llvm/16.0.6/include/llvm-c`, but maybe this directory is not being checked for some reason. 

Any help would be greatly appreciated.

**Comment by marcusrossel** [comment]
Hi, thanks for sharing this! The following steps worked for me:
1. installed LLVM 16 via Homebrew
2. ran `swift make-pkgconfig.swift` (which is also described in the blog post you linked to)
3. cloned this repo
4. updated `Package.swift` to the code snippet shown at the bottom of this comment
5. updated `umbrella.h` to the code snippet shown at the bottom of this comment

Then the project should build in Xcode but give you an error on any reference to `LLVMBuildCall`.
This function [has apparently been deprecated](https://releases.llvm.org/16.0.0/docs/ReleaseNotes.html#changes-to-the-c-api) and replaced with `LLVMBuildCall2`. If you need any help figuring out how to use `LLVMBuildCall2`, or if these steps don't fix your build issue let me know!

```swift
// swift-tools-version:5.7.1

import PackageDescription

let package = Package(
    name: "Kaleidoscope",

    products: [
        .executable(
            name: "Kaleidoscope",
            targets: ["Kaleidoscope"]
        ),
    ],

    dependencies: [ ],

    targets: [
        .systemLibrary(
            name: "CLLVM",
            pkgConfig: "cllvm"
        ),
        .target(
            name: "KaleidoscopeLib",
            dependencies: ["CLLVM"]
        ),
        .target(
            name: "Kaleidoscope",
            dependencies: ["KaleidoscopeLib"]
        ),
        .testTarget(
            name: "KaleidoscopeLibTests",
            dependencies: ["KaleidoscopeLib"]
        ),
    ]
)
```

```c
#define _GNU_SOURCE // https://stackoverflow.com/q/5582211/3208492
#define __STDC_CONSTANT_MACROS // https://stackoverflow.com/a/986584/3208492
#define __STDC_FORMAT_MACROS // https://stackoverflow.com/q/8132399/3208492
#define __STDC_LIMIT_MACROS // https://stackoverflow.com/a/986584/3208492

#include <llvm-c/Analysis.h>
#include <llvm-c/BitReader.h>
#include <llvm-c/BitWriter.h>
#include <llvm-c/Comdat.h>
#include <llvm-c/Core.h>
#include <llvm-c/DataTypes.h>
#include <llvm-c/DebugInfo.h>
#include <llvm-c/Deprecated.h>
#include <llvm-c/Disassembler.h>
#include <llvm-c/DisassemblerTypes.h>
#include <llvm-c/Error.h>
#include <llvm-c/ErrorHandling.h>
#include <llvm-c/ExecutionEngine.h>
#include <llvm-c/ExternC.h>
#include <llvm-c/IRReader.h>
#include <llvm-c/Initialization.h>
#include <llvm-c/LLJIT.h>
#include <llvm-c/Linker.h>
#include <llvm-c/Object.h>
#include <llvm-c/Orc.h>
#include <llvm-c/OrcEE.h>
#include <llvm-c/Remarks.h>
#include <llvm-c/Support.h>
#include <llvm-c/Target.h>
#include <llvm-c/TargetMachine.h>
#include <llvm-c/Transforms/IPO.h>
#include <llvm-c/Transforms/InstCombine.h>
#include <llvm-c/Transforms/PassBuilder.h>
#include <llvm-c/Transforms/PassManagerBuilder.h>
#include <llvm-c/Transforms/Scalar.h>
#include <llvm-c/Transforms/Utils.h>
#include <llvm-c/Transforms/Vectorize.h>
#include <llvm-c/Types.h>
#include <llvm-c/blake3.h>
#include <llvm-c/lto.h>
```

**Comment by lucfis** [comment]
Thank you very much for the help; I got it working. Turns out I was just being an idiot when it came to the pkg-config path, which was why none of the headers were being found. Regardless, I appreciate the updated header names you provided, as copying them over saved me a bit of time.

Switching over to LLVMBuildCall2 was pretty easy thanks to [this](https://discourse.llvm.org/t/llvmbuildcall2-function-type/71093).
